### DISCUSSION TITLE
The 'search' request should have mandatory fields for textual addresses and distance between pickup and drop Location

### DISCUSSION THREAD

**Discussion by tarung** [ONDC-Official/mobility-specification#18] [General]
Currently we only have GPS coordinates for the two locations i.e. 

 "fulfillment": {
        "stops": [
          {
            "type": "START",
            "location": {
              "gps": "13.008935, 77.644408"
            }
          },
          {
            "type": "END",
            "location": {
              "gps": "12.971186, 77.586812"
            }
          }
        ]
      },

Adding  these fields here would be beneficial - textual representation of pickup and drop locations and the estimated distance between pickup locations. 
This addition will help avoid the need for distance calculations between the pickup and drop locations, thus minimizing unnecessary overheads. The underlying assumption here is that the passenger interface already allows users to check the accuracy of the two location selected by the user by showing the textual representation of the selected locations, and another way is to also show the distance between the two locations. So, while sending search request the passenger side can also provide the textual address and estimated distance, this allows the provider side to reduce overhead costs of distance calculations and geocoding the locations.

**Comment by balajer** [comment]
@tarung interesting point. Can you please share a real-world example of how this would look like for the user?

**Reply by tarung** [reply]
Sure, I think the search request body should look similar to this :

"fulfillment": {
        "stops": [
          {
            "type": "START",
            "location": {
              "gps": "13.008935, 77.644408",
              "address" : "2nd Cross Road, Nrupatunga Extension, Dasappa Garden, Tumakuru, Karnataka. Pin-572103 (India)"
            }
          },
          {
            "type": "END",
            "location": {
              "gps": "12.971186, 77.586812"
              "address" : "Saraswathipuram,Tumakuru, Karnataka, 572105"
            }
          }
        ],
	"distance": 1.45,
      },

**Reply by balajer** [reply]
Thanks, Tarun

@92shreyansh - Thoughts?

**Reply by tarung** [reply]
I noticed that address is available.
address | string The address of this location.

but there is no provision of sending total distance.

**Reply by 92shreyansh** [reply]
This option was evaluated and we can definetely consider adding illustartion to showcase how BAP can send address in search and also going forward.

However this would be in consideration that the BPP would not validate the gps and address at there end and GPS still remains sole point of truth for BPP. Adding to that the distance calculation based on GPS and route should be at BPP end only as this is a factor of fare calculation and element to BPPs service. Hence sending a distance and route would not add much value as BPP would definetely like to re-calculate this at there end to provide a accurate fare for the ride.
